src/components/selectionBoxLayer.ts

```diff
@@ -214,6 +273,17 @@ export module Components {
         this.yScale().offUpdate(this._adjustBoundsCallback);
       }
     }
+
+    protected _setBoundsWithoutChangingMode(bounds: Bounds) {
```

Comment:
just calling `_setBounds()` does not change the mode either. The name of this method smells. (actually the existance of three different methods for setting bounds smells)